vllm/sampling_params.py

```diff
@@ -95,6 +95,7 @@ def __init__(
         self.frequency_penalty = frequency_penalty
         self.temperature = temperature
         self.top_p = top_p
+        assert top_k <= 1024
```

Comment:
I got this error when trying `examples/llm_engine_example.py`:
```
  File "/home/wskwon/workspace/vllm/vllm/worker/worker.py", line 89, in profile_num_available_blocks
    sampling_params = SamplingParams(top_p=0.99, top_k=vocab_size - 1)
  File "/home/wskwon/workspace/vllm/vllm/sampling_params.py", line 98, in __init__
    assert top_k <= 1024, top_k
AssertionError: 50271
```
This is because we sample with `top_k=vocab_size - 1` during the initial memory profiling. Could you fix the logic here so that we can use the original implementation when `top_k` is larger than 1024?

Reply:
Sure, but a larger top-k value will result in greater GPU memory consumption.

Reply:
Unfortunately, when I attempted to increase the parameters to over 1024, the new operator produced incorrect results, and I haven't been able to identify the cause of this phenomenon in a short period of time. The[ Fastertransformer](https://github.com/NVIDIA/FasterTransformer/blob/afdf9a9eb86f15363c0249117d166d6b45dbb371/src/fastertransformer/kernels/sampling_topk_kernels.cu) and [TensorRT-LLM](https://github.com/NVIDIA/TensorRT-LLM) I referenced  also doesn't support sampling operators with a topk larger than 1024, and [I'm not sure why](https://github.com/NVIDIA/TensorRT-LLM/issues/61). On the other hand, I would also like you to consider limiting the topk, as excessively large topk values don't have much practical significance and can consume too much GPU memory, slowing down AI inference speed.

Reply:
@WoosukKwon  Sorry to bother you, may I know your views on this issue and hope you can provide some suggestions

Reply:
Can we just treat -1 or 0 as the noop value in the kernel (return early) and set that instead of `vocab_size - 1` during profiling and when top_k is not set by the user? We would also need to reject top_k values above 1024.

Reply:
@WoosukKwon Any update?